core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -75,11 +75,11 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
     * every time off the full set of logs to allow logs to be dynamically added to the pool of logs
     * the log manager maintains.
     */
-  def grabFilthiestLog(): Option[LogToClean] = {
+  def grabFilthiestCompactedLog(): Option[LogToClean] = {
     inLock(lock) {
       val lastClean = allCleanerCheckpoints()
       val dirtyLogs = logs.filter {
-        case (topicAndPartition, log) => log.config.compact  // skip any logs marked for delete rather than dedupe
+        case (topicAndPartition, log) => log.config.compact  // match logs that are marked as compacted
```

Comment:
If one of the variables in the pattern match are not being used, replace it with `_`.